packages/fxa-settings/src/components/FormPasswordWithBalloons/index.tsx

```diff
@@ -0,0 +1,208 @@
+        <div className="relative">
+          <FtlMsg
+            id="form-reset-password-with-balloon-new-password"
+            attrs={{ label: true }}
+          >
+            <InputPassword
+              name="newPassword"
+              className="text-start"
+              label={newPasswordLabel || 'New password'}
+              onFocusCb={onFocusMetricsEvent ? onNewPwdFocus : undefined}
+              onBlurCb={isNewPwdBalloonVisible ? onNewPwdBlur : undefined}
+              onChange={() => {
+                trigger(['newPassword', 'confirmPassword']);
+              }}
+              hasErrors={
+                formState.dirtyFields.newPassword ? errors.newPassword : false
+              }
+              inputRef={register({
+                required: true,
+                validate: {
+                  length: (value: string) => value.length > 7,
+                  notEmail: (value: string) => {
+                    return !passwordValidator.isSameAsEmail(
+                      value.toLowerCase()
+                    );
+                  },
+                  uncommon: async (value: string) => {
+                    // @ts-ignore
+                    const list = await import('fxa-common-password-list');
+                    const input = value.toLowerCase();
+                    return (
+                      !list.test(input) && !passwordValidator.isBanned(input)
+                    );
+                  },
+                },
+              })}
+              prefixDataTestId="new-password"
+            />
+          </FtlMsg>
+          {isNewPwdBalloonVisible && (
+            <PasswordStrengthBalloon
+              {...{
+                hasUserTakenAction: formState.dirtyFields.newPassword,
+                isTooShort: errors.newPassword?.types?.length,
+                isSameAsEmail: errors.newPassword?.types?.notEmail,
+                isCommon: errors.newPassword?.types?.uncommon,
+              }}
+            />
+          )}
+        </div>
+
+        <div className="relative">
+          <FtlMsg
+            id="form-reset-password-with-balloon-confirm-password"
+            attrs={{ label: true }}
+          >
+            <InputPassword
+              name="confirmPassword"
+              label={confirmPasswordLabel || 'Re-enter password'}
+              className="text-start"
+              onFocusCb={showConfirmPwdBalloon ? onConfirmPwdFocus : undefined}
+              onBlurCb={
+                showConfirmPwdBalloon && isConfirmPwdBalloonVisible
+                  ? onConfirmPwdBlur
+                  : undefined
+              }
+              onChange={() => {
+                if (passwordMatchErrorText) {
+                  setPasswordMatchErrorText(undefined);
+                }
+                trigger(['newPassword', 'confirmPassword']);
+              }}
+              // Password confirmation field is disabled until new password is valid
```

Comment:
Thanks for spotting these differences! Here's where we stand on these changes:

> typing allowed in confirmPassword field even if newPassword has errors

FIXED

> If password requirements aren't met, the "Password requirements" balloon should stay open even when you click away from the input.

FIXED:  PasswordInfoBalloon now only shows if the newPassword hasn't been touched or is valid

> When you click either password eye, **both** inputs should show/hide.

FIXED: Added paired visibility buttons as an option in the existing InputPassword component.

> In prod, we can click "Create account" and it sends user focus to the first input box to show that's where the error is. Here, it's just disabled until conditions are met. I'm not sure which is better.

FIXED: I think it's best to return focus to the input that needs to be corrected.

> When password balloons are hidden, there should be a timeout and the balloon should fade-out instead of abruptly hiding

FIXED: Added a timeout - I'm not sure if my solution is the most elegant, **it could benefit from a review.**

> The "passwords don't match" tooltip doesn't show, but, I think we had a separate ticket for that or said we'd do it with the functionality ticket

SOMEWHAT FIXED: The tooltips now show for "Passwords don't match" and "Age is required", but they appear on blur rather than on attempted form submission.

Reply:
<img src="https://media2.giphy.com/media/detOmeyQVP0D1uCX1F/giphy.gif?cid=790b7611e8f3f18aeae22ee78119a09774135d50ee47c826&rid=giphy.gif&ct=g" width="300">

continued convo [here](https://github.com/mozilla/fxa/pull/14781/files#r1092589909).